include/seqan3/core/metafunction/iterator.hpp

```diff
@@ -113,6 +113,66 @@ struct size_type<it_t>
 
 // see specialisation for ranges in core/metafunction/range.hpp
 
+// ----------------------------------------------------------------------------
+// iterator_category
+// ----------------------------------------------------------------------------
+
+/*!\brief Type metafunction that deduces the `iterator_category` from the the modelled concept [Type metafunction].
+* \tparam it_t The type to operate on.
+*
+* | Modelled concept                                                         | iterator_tag<it_t>::type        |
+* |--------------------------------------------------------------------------|---------------------------------|
+* | -                                                                        | not set                         |
+* | std::InputIterator<it_t>                                                 | std::input_iterator_tag         |
+* | !std::InputIterator<it_t> && std::OutputIterator<it_t, value_type<it_t>> | std::output_iterator_tag        |
+* | std::ForwardIterator<it_t>                                               | std::forward_iterator_tag       |
+* | std::BidirectionalIterator<it_t>                                         | std::bidirectional_iterator_tag |
+* | std::RandomAccessIterator<it_t>                                          | std::random_access_iterator_tag |
+*/
+template <typename it_t>
+struct iterator_tag{};
+
+//!\cond
+template <std::InputIterator it_t>
+struct iterator_tag<it_t>
+{
+    using type = std::input_iterator_tag;
+};
+
+template <typename it_t>
+    requires !std::InputIterator<it_t> && std::OutputIterator<it_t, value_type_t<it_t>>
+struct iterator_tag<it_t>
+{
+    using type = std::output_iterator_tag;
+};
+
+template <std::ForwardIterator it_t>
+struct iterator_tag<it_t>
+{
+    using type = std::forward_iterator_tag;
+};
+
+template <std::BidirectionalIterator it_t>
+struct iterator_tag<it_t>
+{
+    using type = std::bidirectional_iterator_tag;
+};
+
+template <std::RandomAccessIterator it_t>
+struct iterator_tag<it_t>
+{
+    using type = std::random_access_iterator_tag;
+};
+//!\endcond
+
+/*!\brief Return the `iterator_category` type of the input type [Type metafunction, shortcut].
+ * \tparam t The type to operate on.
+ */
+template <typename it_t>
+//!\cond
+    requires requires { typename iterator_tag<it_t>::type; }
+//!\endcond
+using iterator_tag_t = typename iterator_tag<it_t>::type;
```

Comment:
add one overload for the shortcut that also works with iterator_category in case the concepts fail and the iterator_traits work.